src/user_lists/views.py

```diff
@@ -49,3 +54,116 @@ def destroy(self, request, *args, **kwargs):
         instance.items.filter(is_removed=False).delete()
         return Response({"success": True}, status=status.HTTP_200_OK)
 
+
+class ListItemViewSet(
+    CreateModelMixin,
+    RetrieveModelMixin,
+    UpdateModelMixin,
+    DestroyModelMixin,
+    ListModelMixin,
+    viewsets.GenericViewSet,
+):
+    queryset = ListItem.objects.filter(is_removed=False)
+    serializer_class = ListItemSerializer
+    permission_classes = [IsAuthenticated, IsObjectOwner]
+
+    def get_queryset(self):
+        queryset = self.queryset.filter(created_by=self.request.user)
+        parent_list_id = self.request.query_params.get("parent_list")
```

Comment:
I think the url param should be `parent_list_id` instead of `parent_list` to be more explicit. We had issues in the past when leaving `_id` part out some parts of the code get confused with assumption of an object instead of ID